plugins/org.python.pydev.ast/src/org/python/pydev/ast/codecompletion/revisited/visitors/LocalScope.java

```diff
@@ -457,6 +459,22 @@ public void setScopeEndLine(int beginLine) {
                 lst.addAll(entry.node.specialsBefore);
             }
 
+            if (entry.node instanceof TryExcept) {
+                TryExcept tryExcept = (TryExcept) entry.node;
+                for (excepthandlerType handle : tryExcept.handlers) {
```

Comment:
The ast must always be checked for `null` (you never know if the ast may be incomplete), so, this must check if `tryExcept.handlers` is null and later on if `handle.name` is null and if `handle.type` is null (if the `handle.type` is null we can't infer its type).